airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -3021,18 +3021,26 @@ def __init__(
         self.polling_interval_seconds = polling_interval_seconds
 
     def execute(self, context: Context):
+        if self.region is None:
+            raise AirflowException("Region should be set here")
         hook = DataprocHook(gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain)
         # batch_id might not be set and will be generated
         if self.batch_id:
+            project_id = self.project_id or hook.project_id
             link = DATAPROC_BATCH_LINK.format(
-                region=self.region, project_id=self.project_id, batch_id=self.batch_id
+                region=self.region, project_id=project_id, batch_id=self.batch_id
             )
             self.log.info("Creating batch %s", self.batch_id)
             self.log.info("Once started, the batch job will be available at %s", link)
```

Comment:
```suggestion
            self.log.info("Once created succesfully, the batch job will be available at %s", link)
```
To somehow mititage the drawback you mentioned in the description

Reply:
Putting the link to the job into logs has been recently removed so I assume that's not needed